app/src/main/java/uk/ac/ebi/atlas/search/metadata/JsonMultiexperimentCellTypeMarkerGenesController.java

```diff
@@ -28,17 +31,26 @@ public JsonMultiexperimentCellTypeMarkerGenesController(
     }
 
     @GetMapping(value = "/json/cell-type-marker-genes/{cellType}",
-                produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
     public String getCellTypeMarkerGenes(
             @PathVariable String cellType,
             @RequestParam(name = "experiment-accessions", required = false) Collection<String> experimentAccessions) {
         return GSON.toJson(
                 highchartsHeatmapAdapter.getMarkerGeneHeatmapDataSortedLexicographically(
                         experimentAccessions == null ?
                                 multiexperimentCellTypeMarkerGenesService.getCellTypeMarkerGeneProfile(
-                                        URLDecoder.decode(cellType, StandardCharsets.UTF_8)) :
+                                        getDecodedCellType(cellType)) :
                                 multiexperimentCellTypeMarkerGenesService.getCellTypeMarkerGeneProfile(
                                         ImmutableSet.copyOf(experimentAccessions),
-                                        URLDecoder.decode(cellType, StandardCharsets.UTF_8))));
+                                        getDecodedCellType(cellType))));
+    }
+
+    private static @NotNull String getDecodedCellType(@NotNull String cellType) {
+        if (cellType == null || cellType.trim().isEmpty()) {
+            throw new IllegalArgumentException("Input cellType cannot be null or empty");
+        }
+
+        byte[] decodedBytes = Base64.getDecoder().decode(cellType);
+        return new String(decodedBytes, StandardCharsets.UTF_8);
```

Comment:
inline variable as we are not using it later
```suggestion
        return new String(Base64.getDecoder().decode(cellType), StandardCharsets.UTF_8);
```